internal/cri/server/service.go

```diff
@@ -285,6 +288,9 @@ func (c *criService) Run(ready func()) error {
 		}()
 	}
 
+	log.L.Info("Starting metric cache provider")
+	c.metricMonitor.Start()
```

Comment:
> Also not clear to me that the conversation in the kubelet issue came to agreement on implementation details.. 

that's fair, it seems this needs more discussion and I will add it to the sig-node agenda again.

> Also if the pod only lives for 9s..

This is something that isn't handled today I think.  The kubelet code has a comment that `If no caller calls
// this function, the cpu usage will stay nil`

> Obviously if the stats are only updated once every 10s.. and they are pulling out of cycle every 10s they could be "behind" by 10s.

The only stat that is "cached" now is the `Usagenanocores` as the [CRI spec](https://github.com/kubernetes/kubernetes/blob/0dc45103d879c3b280671f009d3f830650903894/staging/src/k8s.io/cri-api/pkg/apis/runtime/v1/api.proto#L1720-L1721) defines this as

>    // Total CPU usage (sum of all cores) averaged over the sample window.
>    // The "core" unit can be interpreted as CPU core-nanoseconds per second.

The "sample window" was hard coded to 10s in kubelet but it isn't defined in the cri definition and because it isn't defined here it is the source of the problem.  It needs to clearer who is responsible for generating the value over the "sample window".